Contributing.md

```diff
@@ -95,10 +95,18 @@ wish there weren't. You can leave `FIXME`s, preferably with an issue number.
 
 You may want to run a version of rustfmt from source code as part of a test or
 hacking on the rustfmt codebase. It's strongly discouraged to install a version
-of rustfmt from source. Instead, run it using `cargo run`, and `--manifest-path`.
+of rustfmt from source.
+
+To run `rustfmt` on a file:
+
+```
+cargo run --bin rustfmt -- path/to/file.rs
+```
+
+If you want to test modified `cargo-fmt`, or run `rustfmt` on the whole project:
 
 ```
-cargo run --bin cargo-fmt -- --manifest-path path/to/project/you/want2test/Cargo.toml
+cargo build --bin rustfmt && RUSTFMT="./target/debug/rustfmt" cargo run --bin cargo-fmt -- --manifest-path path/to/project/you/want2test/Cargo.toml
```

Comment:
I'm fine with this as is (I have run into the current non-ideal behavior myself on occasion, but I haven't heard of anyone else doing so while developing rustfmt)

However, this line is starting to get a bit long. What if we went with the one-liner as your original proposal and just included a note above or below (something to the effect of "you may need to build rustfmt first)"

The audience for these docs are strictly rustfmt developers, and I'm content structuring the docs under the assumption they already know/can easily determine this is one binary invoking another and that a build might be involved